vm/src/bytesinner.rs

```diff
@@ -340,24 +340,29 @@ impl PyBytesInner {
         vm: &VirtualMachine,
     ) -> PyResult<()> {
         if let Some(idx) = self.elements.get_pos(int) {
-            let result = match_class!(match object {
-                i @ PyInt => {
-                    if let Some(value) = i.borrow_value().to_u8() {
-                        Ok(value)
-                    } else {
-                        Err(vm.new_value_error("byte must be in range(0, 256)".to_owned()))
-                    }
-                }
-                _ => Err(vm.new_type_error("an integer is required".to_owned())),
-            });
-            let value = result?;
+            let value = Self::value_try_from_object(vm, object)?;
             self.elements[idx] = value;
             Ok(())
         } else {
             Err(vm.new_index_error("index out of range".to_owned()))
         }
     }
 
+    fn value_try_from_object(vm: &VirtualMachine, object: PyObjectRef) -> PyResult<u8> {
+        let value = vm.to_index(&object).ok_or_else(|| {
+            vm.new_type_error(format!(
+                "'{}' object cannot be interpreted as an integer",
+                object.class().name
+            ))
+        })?;
+        // __index__ returned non-int type
+        let value = value?;
```

Comment:
is line 359 needed to be seperated?
```suggestion
        })??;
        // __index__ returned non-int type
```

Reply:
I'd like to spec what could return from the function. Or we just merge it together.
```suggestion
        })??;
```